app/src/main/java/com/droidhats/campuscompass/views/MapFragment.kt

```diff
@@ -82,7 +83,13 @@ class MapFragment : Fragment(), OnMapReadyCallback, GoogleMap.OnMarkerClickListe
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-        return inflater.inflate(R.layout.map_fragment, container, false)
+        val mapFragment = inflater.inflate(R.layout.map_fragment, container, false)
+        requireActivity().onBackPressedDispatcher.addCallback(object : OnBackPressedCallback(true) {
+            override fun handleOnBackPressed() {
+                //do nothing on purpose - user can't go back to splash screen
```

Comment:
we should be exiting the application here actually. We don't want them to be stuck inside the application...